src/app/components/client/PremiumBadge.tsx

```diff
@@ -56,8 +51,31 @@ export default function PremiumBadge({
     { type: "dialog" },
     dialogState,
   );
+  return (
+    <>
+      <Button {...triggerProps} variant="primary" small>
+        {props.label}
+      </Button>
+      <PremiumUpsellDialog
+        {...overlayProps}
+        state={dialogState}
+        monthlySubscriptionUrl={props.monthlySubscriptionUrl}
+        yearlySubscriptionUrl={props.yearlySubscriptionUrl}
+      />
+    </>
+  );
+};
+
+export function PremiumButton(props: Props) {
+  return <PremiumLayout {...props} />;
+}
```

Comment:
That’s perfect, thanks! What I meant was: Why not just rename `PremiumLayout` to `PremiumButton` and use that directly instead of the additional wrapper?